solidity/contracts/BTCUtils.sol

```diff
@@ -137,20 +157,29 @@ library BTCUtils {
     /// @param _vin      The vin as a tightly-packed byte array
     /// @param _index    The 0-indexed location of the input to extract
     /// @return          The input as a byte array
-    function extractInputAtIndex(bytes memory _vin, uint8 _index) internal pure returns (bytes memory) {
-        uint256 _len;
+    function extractInputAtIndex(bytes memory _vin, uint256 _index) internal pure returns (bytes memory) {
+        uint256 _varIntDataLen;
+        uint256 _nIns;
+
+        (_varIntDataLen, _nIns) = parseVarInt(_vin);
+        require(_varIntDataLen != ERR_BAD_ARG, "Read overrun during VarInt parsing");
+        require(_index <= _nIns, "Vin read overrun");
```

Comment:
Note that these requires can be bypassed if `determineVarIntDataLength(_vin)` is zero and `_index` is zero.

Reply:
slight bug here. 166 should be `<` instead of `<=`. same for L343 in `extractOutputAtIndex`